riscv/simif.h

```diff
@@ -19,10 +24,17 @@ class simif_t
   // Callback for processors to let the simulation know they were reset.
   virtual void proc_reset(unsigned id) = 0;
 
+  virtual processor_t* get_core(size_t i) = 0;
+  virtual const cfg_t &get_cfg() = 0;
+  virtual unsigned nprocs() const = 0;
+  virtual const std::map<size_t, processor_t*>& get_harts() = 0;
+
```

Comment:
I think these can be deduplicated so we can have a narrower interface for `simif_t`. For example, it looks like debug_module only uses `get_cfg` to get `nprocs` and `hartids`.

Reply:
Longer term, I want to get rid of `get_core`, but AFAICS its only usage is inside of `sim_t` itself.  So we can avoid adding it to the `simif_t` interface.

I have a feeling `get_cfg` will keep rearing its head, even though you appear to be right about its current usage.

`nprocs` is just a convenience.  Rather than making it virtual, we could define it here as a non-virtual method that returns `get_cfg().nprocs()`.

Reply:
I took out `get_core` from `simif_t` (not sure why I put it there in the first place) and implemented `nprocs()` inside `simif_t`.